File: Processus/exercice2.1.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/wait.h>

int main()
{
    pid_t pid ;
    pid = fork() ;

    if(pid == -1)
    {
        perror("Erreur lors de la création du processus ") ;
        exit(1) ;
    }
    else if(pid == 0)
    {
        printf("Je suis fils \n") ;
    }
    else
    {
        printf("Je suis le père \n") ;
    }
}

File: Processus/exercice2.1.py
import os
import sys

def main():
    try:
        pid = os.fork()
    except OSError:
        print("Erreur lors de la création du processus")
        sys.exit(1)

    if pid == 0:
        print("Je suis fils")
    else:
        print("Je suis le père")

if __name__ == "__main__":
    main()